### ISSUE TITLE
README pages for new python lecture repos

### ISSUE THREAD

**Issue by mmcky** [QuantEcon/lecture-python-website#57]
Currently the `README` files for each lecture repo is empty. 

@jstac would you like the old style guides etc. replicated across all three repos? 

Given it will likely be very similar across the three should we update the `README` to this `meta` repo and then add links to the others for Style Guide etc. to the README located in `lecture-python-website`?

**Comment by mmcky** [comment]
@jstac Do you think we should setup docs to put together a writing guide and then we can link to a single location from all three REPOS. We could host the docs in this `meta` python website repo.

**Comment by jstac** [comment]
Yes, nice idea.  We have these somewhat neglected pages: https://quantecon.org/contribute/

We could put the info in the README there, as a guide for contributing to the lectures (as opposed to the libraries).

That could be combined with a review of those pages.  When reviewing, one aim should be to cut whatever information from our site we can easily outsource to somewhere else, and generally reduce the number of words.  @Harveyt47 and @sayaikegawa might be willing to have a go at this.

**Comment by mmcky** [comment]
Thanks -- this is a great idea. Let's make use of the `contribute` page for Lectures.

**Comment by mmcky** [comment]
@Harveyt47 could you let me know when you have a PR ready for this. thanks.

**Comment by Harveyt47** [comment]
@mmcky yep, @sayaikegawa and I have discussed the following changes 

- move 'Develop and contribute code' [section](https://quantecon.org/contribute/
) to [QuantEcon.py](https://quantecon.org/quantecon-py/) and [QuantEcon.jl](https://quantecon.org/quantecon-jl/) project pages
- Move 'Write a notebook' [section](https://quantecon.org/contribute/) to QuantEcon-notes about [page](https://notes.quantecon.org/about)

We talked about moving the git resources to somewhere else but have not found a place yet.

**Comment by mmcky** [comment]
@Harveyt47 @sayaikegawa @jstac 

I think we need to build out the `contributions` pages rather than just a direct replacement. @jstac please chime in here if you disagree. I think the quantecon.org/contribute.html should connect people to contributing to all of `quantecon` projects (i.e. `lectures`, `software` etc. etc.) with Lectures being the most important. So I would think a nice top level page would link to

quantecon.org/contribute-lectures.html
quantecon.org/contribute-software.html

etc. 

I think the style guide would go on quantecon.org/contribute-lectures.html and be linked to from the lecture repositories. 

This is a nice unified location to document all ways people can contribute in the various ways but I agree its wordy and needs to be cut down into basically perhaps a badge style page?

**Comment by Harveyt47** [comment]
@mmcky I like that idea.

**Comment by mmcky** [comment]
I am thinking of badges for:

1. Lectures (contain Style Guide for writing, and PRs)
1. Python Projects
1. Julia Projects (as Julia team to submit a contributors guide)
1. Help Others (Discourse Forum etc)
1. QuantEcon Notes
1. (optional) Executable Book Project Contributors Page but thought it would be nice to embed link to that project
1. Donations

**Comment by mmcky** [comment]
@DrDrij might you be able to assist with the intermediate `contribution` landing page?

@Harveyt47 can you start work on:

1. contribute-lectures
1. contribute-python-software

with migration of information that we currently have (with any improvements you can think of)

**Comment by Harveyt47** [comment]
@mmcky  contribute-python-software is for contributing to QuantEcon.py yeah?.

If so, we already have this [page](https://quantecon.org/python-developers/) about how to contribute. Same for [julia](https://quantecon.org/julia-developers/).

**Comment by mmcky** [comment]
@Harveyt47 we can start with that but it would be good to have a general set of guidelines for contributing to `python` projects (i.e. similar testing infrastructure etc.)

**Comment by Harveyt47** [comment]
Yep ok sounds good.

**Comment by mmcky** [comment]
@Harveyt47 as part of the writeup for `contribute-python` page perhaps you can also try and setup `Github Actions` on QuantEcon.py via a PR. You can reference the [travis.yml](https://github.com/QuantEcon/QuantEcon.py/blob/master/.travis.yml) file for our current configuration and then setup `.github/workflows`. 

Please use [Github Actions docs](https://help.github.com/en/actions) and look at `.github/worfklows` in other QuantEcon software projects such as [jupyter-book](https://github.com/executablebooks/jupyter-book/tree/master/.github/workflows) -- and then ping me on slack if you get stuck.

**Comment by Harveyt47** [comment]
Hey @mmcky I'm reading up on GitHub actions, looks interesting, I'll ping you slack when I have questions. Thanks.

**Comment by mmcky** [comment]
hey @Harveyt47 when you have a chance -- can you update me on this migration of content?

**Comment by Harveyt47** [comment]
Hey @mmckay I have migrated the README content into a `contribute_lectures.md` file. I will link this page to the ‘Contribute to the lecture’s section on the Contribute page. I will make a PR for this in the QuantEcon/website/pages repo.

I Have come up with a GitHub Actions ci with `nose` test that works successfully. I have also explained some of the test peoples code will go through and expectations like `PEP8`. This Will be added to the Contribute to QuantEcon.py page. As far as general guideline for contributing the `QuantEcon.py` is there anything you had in mind that you want me to add. 

I am still working on the `pytest`, this will require some changes to the tests. I have implemented the needed changes locally and can get the tests working. Also, with the changing of tests to `pytest` we will have to update some sections of the Contribute to QuantEcon.py page and linked pages. But I think I’ll leave that until it has been implemented in Github Actions.

**Comment by mmcky** [comment]
> I will make a PR for this in the QuantEcon/website/pages repo.
Great. Thanks. 

> As far as general guideline for contributing the QuantEcon.py is there anything you had in mind that you want me to add.
It is mainly PEP8, testing requirements etc. We could add some additional requirements if we like such as `flake8`, `black` but I find them onerous for new contributors and that may deter contributors to QuantEcon.py so a bit reluctant to introduce them. 

Thanks for the update @Harveyt47 feel free to open PR's when you're ready and I can review etc.

**Comment by mmcky** [comment]
@Harveyt47 has added a page for contributing to lectures which contains the old README information when we had a unified lecture site. This page is now hosted on the `orgsite` in [QuantEcon/website](https://github.com/QuantEcon/website) . 

There are still quite a few elements to this issue. 

@DrDrij could you let me know when you might get a chance to mock up a page (as suggested above) for contributors. Thanks.
